concept/type_/type_manager.rs

```diff
@@ -772,14 +993,26 @@ impl<Snapshot: WritableSnapshot> TypeManager<Snapshot> {
         definition_key: DefinitionKey<'static>,
         field_name: String,
     ) -> Result<(), ConceptWriteError> {
-        // TODO: Validation
-        let mut struct_definition = TypeReader::get_struct_definition(snapshot, definition_key.clone())
-            .map_err(|source| ConceptWriteError::ConceptRead { source })?;
+        let mut struct_definition = TypeReader::get_struct_definition(snapshot, definition_key.clone())?;
         struct_definition.delete_field(field_name).map_err(|source| ConceptWriteError::Encoding { source })?;
+
         TypeWriter::storage_put_struct(snapshot, definition_key.clone(), struct_definition);
         Ok(())
     }
 
+    pub fn delete_struct(
+        &self,
+        snapshot: &mut Snapshot,
+        definition_key: &DefinitionKey<'static>,
+    ) -> Result<(), ConceptWriteError> {
+        // TODO: Check that struct is not referred from attributes (value type) and other structs (fields)!
+        // OperationTimeValidation::validate_struct_is_not_used(snapshot, entity_type.clone().into_owned())
+        //     .map_err(|source| ConceptWriteError::SchemaValidation { source })?;
```

Comment:
I wanted the current tests to pass (otherwise it's hard to control that overall the tests are written correctly as they always change), but your idea makes sense for this place as it could be done just through the schema level. Will do